File: app/src/main/java/fr/piotr/dismoitoutsms/messages/Message.java
package fr.piotr.dismoitoutsms.messages;

import android.support.annotation.NonNull;

import java.util.Date;

import fr.piotr.dismoitoutsms.contacts.Contact;

public class Message implements Comparable<Message> {

	private Date	date;
	private Contact	contact;
	private String	message;

	public Message(Date date, Contact contact, String message) {
		this.date = date;
		this.contact = contact;
		this.message = message;
	}

	public boolean contactIs(Contact contact) {
		return this.contact.equals(contact);
	}

	public void append(String message) {
		this.message += " " + message;
	}

	public Contact getContact() {
		return contact;
	}

	public String getMessage() {
		return message;
	}

	public Date getDate() {
		return date;
	}

	public int compareTo(@NonNull Message another) {
		return this.date.compareTo(another.date);
	}
}

File: app/src/main/java/fr/piotr/dismoitoutsms/messages/Message.kt
package fr.piotr.dismoitoutsms.messages

import fr.piotr.dismoitoutsms.contacts.Contact
import java.io.Serializable
import java.util.*

data class Message(val smsId: Long = -1,
                   val threadId: Long = -1,
                   var messageType: MessageType,
                   val date: Date,
                   val contact: Contact,
                   var message: String,
                   var read: Boolean = false,
                   var errorCode: Int? = null,
                   val discriminant: Discriminant = Discriminant.SMS,
                   val messageParts: List<MessagePart> = listOf()) : Comparable<Message>, Serializable {

    enum class Discriminant {
        SMS, MMS
    }

    enum class MessageType {
        INBOX, SENT, DRAFT
    }

    fun contactIs(contact: Contact): Boolean {
        return this.contact == contact
    }

    fun append(message: String) {
        this.message += " $message"
    }

    override fun compareTo(other: Message): Int {
        return this.date.compareTo(other.date)
    }
}
